docs: add testing guide for Tegel and link in README (#854)
scania-digital-design-system/tegel: docs/TESTING_GUIDE.md

# Testing Guide for Tegel

## Overview

This guide provides developers with best practices and guidelines on how and when to write unit tests for components within our design system. Our aim is to ensure that each component is reliable, maintainable, and functions as intended.

---

## How to Unit Test Components

### Testing Setup

- We use **Playwright** for unit testing.

> - Each component should have its own test file located in the same folder as the component (`componentName.e2e.ts`).

Reviewer: ```suggestion
- Each component should have its own test file located in the same folder as the component (`../components/componentName/test/componentName.e2e.ts`).
```

# Testing Guide for Tegel

## Overview

This guide provides developers with best practices and guidelines on how and when to write unit tests for components within our design system. Our aim is to ensure that each component is reliable, maintainable, and functions as intended.

---

## How to Unit Test Components

### Testing Setup

- We use **Playwright** for unit testing.

> - Each component should have its own test file located in the same folder as the component (`componentName.e2e.ts`).

Reviewer: ```suggestion
- Each component should have its test file located in the folder of the component (`../components/button/test/button.e2e.ts`).
```

Reviewer: Also, we used folders to specify what kind of test is conducted, for example: disabled, danger etc... so usually more complex component have a subfolders that indicate specific test area.
Take a look on button component in code and see what is the best way to present this to the user.
